lib/Screens/EditProfile/edit_profile_screen.dart

```diff
@@ -0,0 +1,424 @@
+        nameController.text = user["name"];
+        userEmail = user["email"];
+      });
+      UserService().getUserApp(user["email"]).then((userApp) {
+        if (userApp.age != null) {
+          ageController.text = userApp.age!;
+        }
+        if (userApp.phone != null) {
+          phoneController.text = userApp.phone!;
+        }
+        if (userApp.occupation != null) {
+          occupationController.text = userApp.occupation!;
+        }
+        if (userApp.workplace != null) {
+          workController.text = userApp.workplace!;
+        }
+        if (userApp.gender != null && userApp.gender != "") {
+          setState(() {
+            genderDropDown = userApp.gender;
+          });
+        }
+        if (userApp.maritalStatus != null && userApp.maritalStatus != "") {
+          setState(() {
+            civilDropDown = userApp.maritalStatus;
+          });
+        }
+      });
+    });
+  }
+
+  saveUserProfile(BuildContext context) async {
+    if (genderDropDown == null) {
+      genderDropDown = "";
+    }
+    if (civilDropDown == null) {
+      civilDropDown = "";
+    }
+    await UserService()
+        .saveUserProfile(
+            userEmail,
+            nameController.text,
+            genderDropDown,
+            ageController.text,
+            workController.text,
+            civilDropDown,
+            occupationController.text,
+            phoneController.text)
+        .then((_) {
+      HelperFunctions.saveUserNameInSharedPreference(nameController.text);
+      showDialog<String>(
+        context: context,
+        builder: (BuildContext context) => AlertDialog(
+          title: const Text('Êxito!'),
+          content: const Text('Seu perfil foi salvo com sucesso!'),
+          actions: <Widget>[
+            TextButton(
+              onPressed: () async {
+                Navigator.of(context)
+                    .popUntil(ModalRoute.withName('/logged-home'));
+                Navigator.of(context).pushNamed("/logged-home");
+              },
+              child: const Text('Ok'),
+            ),
+          ],
+        ),
+      );
+    }).catchError((error) {
+      showDialog<String>(
+        context: context,
+        builder: (BuildContext context) => AlertDialog(
+          title: const Text('Erro!'),
+          content: Text(error.toString()),
+          actions: <Widget>[
+            TextButton(
+              onPressed: () async {
+                Navigator.pop(context);
+              },
+              child: const Text('Ok'),
+            ),
+          ],
+        ),
+      );
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: AppColors.green,
+        shadowColor: Colors.transparent,
+        title: Text("Editar perfil"),
+        leading: IconButton(
+          icon: Icon(Icons.arrow_back),
+          onPressed: () {
+            Navigator.of(context).pushNamed("/logged-home");
```

Comment:
Cria função para todos os onPressed, tem função repetida ai

Reply:
Todas as funções são diferentes entre si

Reply:
sim, eh para uma melhor vizualizacao e poder reutilizar qdo for igual